src/v/rpc/reconnect_transport.cc

```diff
@@ -35,40 +37,44 @@ ss::future<> reconnect_transport::stop() {
     return _dispatch_gate.close().then([this] { return _transport.stop(); });
 }
 
-ss::future<result<transport*>> reconnect_transport::get_connected() {
+ss::future<result<transport*>> reconnect_transport::get_connected(
+  rpc::clock_type::time_point connection_timeout) {
     if (is_valid()) {
         return ss::make_ready_future<result<transport*>>(&_transport);
     }
-    return reconnect();
+    return reconnect(connection_timeout);
 }
 
-ss::future<result<transport*>> reconnect_transport::reconnect() {
+ss::future<result<transport*>>
+reconnect_transport::reconnect(rpc::clock_type::time_point connection_timeout) {
     using ret_t = result<transport*>;
     if (!has_backoff_expired(
           _stamp, _backoff_policy.current_backoff_duration())) {
         return ss::make_ready_future<ret_t>(errc::exponential_backoff);
     }
     _stamp = rpc::clock_type::now();
-    return with_gate(_dispatch_gate, [this] {
-        return with_semaphore(_connected_sem, 1, [this] {
+    return with_gate(_dispatch_gate, [this, connection_timeout] {
+        return with_semaphore(_connected_sem, 1, [this, connection_timeout] {
             if (is_valid()) {
                 return ss::make_ready_future<ret_t>(&_transport);
             }
-            return _transport.connect().then_wrapped([this](ss::future<> f) {
-                try {
-                    f.get();
-                    rpclog.debug(
-                      "connected to {}", _transport.server_address());
-                    _backoff_policy.reset();
-                    return ss::make_ready_future<ret_t>(&_transport);
-                } catch (...) {
-                    _backoff_policy.next_backoff();
-                    rpclog.trace(
-                      "error reconnecting {}", std::current_exception());
-                    return ss::make_ready_future<ret_t>(
-                      errc::disconnected_endpoint);
-                }
-            });
+            vlog(rpclog.debug, "connecting to {}", _transport.server_address());
+            return _transport.connect(connection_timeout)
+              .then_wrapped([this](ss::future<> f) {
+                  try {
+                      f.get();
+                      rpclog.debug(
```

Comment:
you changed this from trace->debug why? if you want to propose this change, please split it into different commits, or revert to use the original. It muddles the intent of this change.